### PR TITLE
Force LF line endings for the formatter

### PR DESCRIPTION
**Pull request by gsmet**
Supersedes https://github.com/quarkusio/quarkus/pull/6474

@gastaldi can you check that the issue you were talking about in the PR above is fixed by this PR. I applied Stuart's change to all projects.

### CODE REVIEW

**Comment by gastaldi** [comment]
@gsmet that doesn't work, I still get the same behavior when building in Windows: 

```
The file will have its original line endings in your working directory
warning: LF will be replaced by CRLF in core/builder/src/main/java/io/quarkus/builder/BuildStep.java
```

**Comment by gastaldi** [comment]
Hold on, let me try again.
I forgot to `git config --global core.autocrlf true`

**Comment by gastaldi** [comment]
Nope, with the `git config --global core.autocrlf true` I still see modified files when I simply  `mvn clean install` the project

**Comment by rquinio** [comment]
This PR works for me.

```
$ git config core.autocrlf
false
$mvn process-sources -f core/builder/
[INFO] --- formatter-maven-plugin:2.11.0:format (default) @ quarkus-builder ---
[INFO] Using 'UTF-8' encoding to format source files.
[INFO] Number of files to be formatted: 30
[INFO] Successfully formatted:          0 file(s)
[INFO] Fail to format:                  0 file(s)
[INFO] Skipped:                         30 file(s)
[INFO] Read only skipped:               0 file(s)
[INFO] Approximate time taken:          2s
```

@gastaldi Did you delete and re-create the files when switching between core.autocrlf settings so changes are reflected in the working directory ?
i.e `git ls-files -z | xargs -0 rm -f` + `git reset --hard HEAD` 

Either way I think the setup should be documented in CONTRIBUTING.md, because the Git Windows installer requires to pick a value that becomes the global git default and there's a 50% chance users picked the wrong one.
i.e. after git clone of Quarkus, Windows users who have `git config core.autocrlf` return `true` should in run in the working directory:
* `git config core.autocrlf false` (local config to the Quarkus repo)
* `git ls-files -z | xargs -0 rm -f`
* `git reset --hard HEAD`

**Comment by gsmet** [comment]
I will merge this one. It's better to have it, even if we think it doesn't solve the problem completely.
